rclpy/test/test_executor.py

```diff
@@ -348,7 +350,23 @@ def test_executor_add_node(self):
         assert not executor.add_node(self.node)
         assert id(executor) == id(self.node.executor)
 
-    def test_executor_spin_until_future_complete_timeout(self):
+    def test_executor_spin_for(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        def timer_callback():
+            pass
+        timer = self.node.create_timer(0.003, timer_callback)
+
+        start = time.monotonic()
+        executor.spin_for(duration_sec=0.1)
+        end = time.monotonic()
+        self.assertGreaterEqual(end - start, 0.1)
```

Comment:
i think this could be too small duration for some platforms might not work as we expect. how about having the following?

```suggestion
        timer = self.node.create_timer(0.1, timer_callback)

        start = time.monotonic()
        executor.spin_for(duration_sec=1.0)
        end = time.monotonic()
        self.assertGreaterEqual(end - start, 1.0)
```